modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -786,11 +792,13 @@ private void sendInputMethodEventToFX(InputMethodEvent e) {
         if (e.getCaret() != null) {
             insertionIndex = e.getCaret().getInsertionIndex();
         }
-        scenePeer.inputMethodEvent(
-                javafx.scene.input.InputMethodEvent.INPUT_METHOD_TEXT_CHANGED,
-                InputMethodSupport.inputMethodEventComposed(t, e.getCommittedCharacterCount()),
-                t.substring(0, e.getCommittedCharacterCount()),
-                insertionIndex);
+        if (scenePeer != null) {
+            scenePeer.inputMethodEvent(
```

Comment:
did we trip over the same problem again here?

which thread modifies the `scenePeer` field?  which thread calls `sendInputMethodEventToFX()` ?